elasticgraph-graphql/lib/elastic_graph/graphql/http_endpoint.rb

```diff
@@ -196,8 +196,8 @@ def normalized_headers
         end
       end
 
-      def content_type
-        normalized_headers["CONTENT-TYPE"]
+      def mime_type
+        @mime_type ||= normalized_headers["CONTENT-TYPE"]&.split(";")&.first # strip ; delimiter and any trailing parameters
```

Comment:
```suggestion
        @mime_type ||= normalized_headers["CONTENT-TYPE"].to_s.split(";").first # strip ; delimiter and any trailing parameters
```

`&.` creates a conditional branch since `foo&.bar` is the equivalent of:

```ruby
if foo != nil
  foo.bar
else
  nil
end
```

Our enforcement of 100% test coverage (outside of `# :nocov:` comments) is failing because the implicit `else` branches aren't covered. But we can short circuit that by using `.to_s` -- `nil.to_s` returns `""` (it's the nice thing about `nil` being an actual object in Ruby compared to `null` in java!) and you can split an empty string just fine:

```
3.4.2 :001 > nil.to_s.split(";").first
 => nil
```

Reply:
thanks, makes sense - I noticed the coverage drop and hadn't gotten to dig into it yet. fixed!